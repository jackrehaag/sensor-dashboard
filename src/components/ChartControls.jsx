import React from "react";

const ChartControls = props => {
  return (
    <div>
      <button
        id="chartControlTemperatureButton"
        className="ui button small"
        type="button"
        value="temperature"
        onClick={props.onMetricSelection}
      >
        <i className="thermometer icon" />
        Temperature
      </button>
      <button
        id="chartControlHumidityButton"
        className="ui button small"
        type="button"
        value="humidity"
        onClick={props.onMetricSelection}
      >
        <i className="tint icon" />
        Humidity
      </button>
      <button
        id="chartControlMotionDetectionButton"
        className="ui button small"
        type="button"
        value="motion_detection"
        onClick={props.onMetricSelection}
      >
        <i className="street view icon" />
        Motion detection
      </button>
      <form
        className="ui form"
        onSubmit={props.onSubmit}
        style={{ display: "inline", padding: ".5833em .833em" }}
      >
        <div className="ui action small input">
          <input
            type="text"
            id="chartControlDateField"
            placeholder="Date"
            onChange={props.onDateChange}
            value={props.dateSelection}
          />
          <button id="chartControlSubmitButton" className="ui button">
            Submit
          </button>
        </div>
      </form>
    </div>
  );
};

export default ChartControls;
