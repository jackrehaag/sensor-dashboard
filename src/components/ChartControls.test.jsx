import React from "react";
import ReactDOM from "react-dom";
import ChartControls from "./ChartControls";
import renderer from "react-test-renderer";
import ReactTestUtils from "react-dom/test-utils";

beforeAll(() => {
  const onMetricSelection = () => {};
  const onSubmit = () => {};
  const onDateChange = () => {};
  const dateSelection = () => {};
});

test("it renders", () => {
  const component = renderer.create(
    <ChartControls dateSelection="2019-06-24" />
  );
  const tree = component.toJSON();
  expect(tree).toMatchSnapshot();
});

describe("callback functions are set", () => {
  const selectionMock = jest.fn();
  const submitMock = jest.fn();
  const dateChangeMock = jest.fn();

  const component = renderer.create(
    <ChartControls
      dateSelection="2019-06-24"
      metricSelection="humidity"
      onSubmit={submitMock}
      onMetricSelection={selectionMock}
      onDateChange={dateChangeMock}
    />
  );

  const root = component.root;

  test("it triggers the metric selection function on button click", () => {
    const humidityButton = root.find(
      element => element.props.value == "humidity"
    );
    humidityButton.props.onClick();
    expect(selectionMock).toHaveBeenCalled();
  });

  test("it triggers the submission callback function form submission", () => {
    const form = root.findByType("form");
    form.props.onSubmit();
    expect(submitMock).toHaveBeenCalled();
  });

  test("it triggers the date change function on updating the date", () => {
    const dateField = root.findByProps(
      element => element.props.id == "chartControlMotionDetectionButton"
    );
    dateField.props.onDateChange();
    expect(selectionMock).toHaveBeenCalled();
  });
});
