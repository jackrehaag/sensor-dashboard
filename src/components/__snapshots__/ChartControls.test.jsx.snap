// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders 1`] = `
<div>
  <button
    className="ui button small"
    id="chartControlTemperatureButton"
    type="button"
    value="temperature"
  >
    <i
      className="thermometer icon"
    />
    Temperature
  </button>
  <button
    className="ui button small"
    id="chartControlHumidityButton"
    type="button"
    value="humidity"
  >
    <i
      className="tint icon"
    />
    Humidity
  </button>
  <button
    className="ui button small"
    id="chartControlMotionDetectionButton"
    type="button"
    value="motion_detection"
  >
    <i
      className="street view icon"
    />
    Motion detection
  </button>
  <form
    className="ui form"
    style={
      Object {
        "display": "inline",
        "padding": ".5833em .833em",
      }
    }
  >
    <div
      className="ui action small input"
    >
      <input
        id="chartControlDateField"
        placeholder="Date"
        type="text"
        value="2019-06-24"
      />
      <button
        className="ui button"
        id="chartControlSubmitButton"
      >
        Submit
      </button>
    </div>
  </form>
</div>
`;
